dvc/exceptions.py

```diff
@@ -206,6 +207,18 @@ def __init__(self, etag, cached_etag):
         )
 
 
+class FileExistsLocallyError(FileExistsError, DvcException):
+    def __init__(self, path, hint=None):
+        import os.path
+
+        self.path = path
+        hint = "" if hint is None else f". {hint}"
+        path_typ = "directory" if os.path.isdir(path) else "file"
+        msg = f"The {path_typ} '{path}' already exists locally{hint}"
+        super().__init__(msg)
+        self.errno = errno.EEXIST
```

Comment:
Setting errno after initializing super to keep the error string nice (no need for "[Errno 17]")
Note that I kept inheritance from `FileExistsError` and the errno for canonical behavior and to maintain backwards compatibility of the api in case people use this to catch those errors. I'm not sure it's needed or important, though 🤷